File: tests/test_php/src/Models/Order.php
<?php

namespace App\Models;

use App\Models\User;

class Order {
    private $id;
    private $userId;
    private $total;

    public function __construct($id, $userId, $total) {
        $this->id = $id;
        $this->userId = $userId;
        $this->total = $total;
    }

    public function getId() {
        return $this->id;
    }

    public function getUserId() {
        return $this->userId;
    }

    public function getTotal() {
        return $this->total;
    }

    public function getOwner(User $user) {
        return $user->getName();
    }

    public function formatTotal() {
        return sprintf("$%.2f", $this->total);
    }
}
?>

File: tests/test_lua/models/order.lua
local User = require("models.user")

local Order = {}
Order.__index = Order

function Order.new(id, userId, amount)
    local self = setmetatable({}, Order)
    self.id = id
    self.userId = userId
    self.amount = amount
    return self
end

function Order:getTotal()
    return self.amount
end

return Order
